### DISCUSSION TITLE
Adding elements in collection of nested signals

### DISCUSSION THREAD

**Discussion by git-noise** [leptos-rs/leptos#2800] [General]
Hello,

I am currently working with some nested signals - something like `Signal<Vec<Signal<Datum>>>` - where `Signal` is the `RwSignal`.
Nested signals are used here because each `Datum` may be updated in the UI and we need to keep track of their potential changes.
Things are working fine as long as we work with the existing collection. Removing some also works.

However, we would also want the ability to add a new `Datum` to the collection.
Issue we are facing, is that whenever we add a `Signal<Datum>` to the collection via a component, it ends up being dropped.
We are currently trying something like that: `data.with(|v| v.push(create_signal(datum)))`.

I may have missed an obvious example in the book, but is there a recommended way to handle adding data to a collection of nested signals?

Thanks a lot for any insight!

**Comment by gbj** [comment]
Not sure what you mean, exactly. If you can provide a minimal compiling example I'm happy to try to figure it out. The `counters` example ([0.6](https://github.com/leptos-rs/leptos/blob/leptos_0.6/examples/counters/src/lib.rs) - [0.7](https://github.com/leptos-rs/leptos/blob/main/examples/counters/src/lib.rs)) is a good example of nested signals.

**Reply by git-noise** [reply]
Thank you for the pointer. I'll have a look and otherwise I'll try and come-up with a reproducible example - right now the code is a tad convoluted.
